lib/std/fs/file.zig

```diff
@@ -8,8 +8,13 @@ const assert = std.debug.assert;
 const windows = os.windows;
 const Os = std.builtin.Os;
 const maxInt = std.math.maxInt;
+const FifoType = std.fifo.LinearFifo(u8, std.fifo.LinearFifoBufferType{ .Static = 4 });
 const is_windows = builtin.os.tag == .windows;
 
+threadlocal var win_surrogate: ?u16 = null;
```

Comment:
Hmm after thinking a bit more about this, I'm not sure this should be `threadlocal` after all. What I'd propose would be:
  - Leave the buffer as a static global (non-threadlocal)
  - Any use of `stdin` across threads on Windows requires you to synchronize I/O access using your own mutex or similar
  
Do you see any issues with that approach? cc @kprotty

Reply:
Its an odd restriction for only a single platform. If it does exist, I think it should extend to all platforms and even all tty Files

Reply:
@topolarity I've been mulling over the implications of `threadlocal` as well. Your suggestion sounds like exactly what the proposal I mentioned before definitely wanted to _maybe_ tackle.

@kprotty  I can try to see what I can cook up to settle this in a platform agnostic way, while trying to dodge the bullet of changing any current public APIs/types.

Reply:
My thought was to require the user to synchronize their use of stdout/stdin with their own mutex, not to wrap this in a mutex in the standard library

That would leave the default behavior fast for single-threaded usage, while still making it possible to perform multi-threaded access correctly (multi-threaded access is broken with threadlocal, since each thread will see a different buffer)